life_expectancy/cleaning.py

```diff
@@ -3,19 +3,34 @@
 """
 
 import re
-import argparse
+from pathlib import Path
+from typing import List
 import pandas as pd
 import numpy as np
 
-def clean_data(country):
-    """Function that will clean, transform and filter (by region, its default being PT) the life_expectancy data."""
-    
-    # Load data
-    dataframe = pd.read_csv("life_expectancy/data/eu_life_expectancy_raw.tsv", sep='\t')
+PROJECT_DIR = Path(__file__).parent
+FILE_PATH = f"{PROJECT_DIR}/data/eu_life_expectancy_raw.tsv"
+SAVE_PATH = "life_expectancy/data/pt_life_expectancy.csv"
+COL_TO_SPLIT = 'unit,sex,age,geo\\time'
+SPLITTED_COLS_LIST = ['unit', 'sex', 'age', 'region']
+
+
+
+def main(country: str = "PT") -> None: #pragma: no cover
+    """Main function."""
+
+    dataframe = load_data(FILE_PATH, '\t')
+
+    cleaned_df = clean_data(dataframe, COL_TO_SPLIT, SPLITTED_COLS_LIST, country)
 
-    # Initialize some vars
-    col_to_split = 'unit,sex,age,geo\\time'
-    splitted_cols_list = ['unit', 'sex', 'age', 'region']
+    save_data(cleaned_df, SAVE_PATH)
+
+def load_data(file_path: str, sep: str = ",") -> pd.DataFrame:
+    """Loads a CSV file given a path and separator."""
+    return pd.read_csv(file_path, sep=sep)
+    
+def clean_data(dataframe: pd.DataFrame, col_to_split: str, splitted_cols_list: List[str], country: str) -> pd.DataFrame:
```

Comment:
**Nitpick:** this function can be considered to have too much variables, even if I don't think it is a real problem, more personal taste.
Since, in my opinion the changes we do to this data end up being very specific, I wouldn't use the variables `col_to_split` and `splitted_cols_list` as input variables, but instead just put them inside of the function

Reply:
nice tip, will implement